### PR TITLE
【Infer Symbolic Shape BUAA No.221】Add sequence_pool op

### PR DESCRIPTION
**Pull request by lwkhahaha**
### PR Category

User Experience 

### PR Types

 Others 

### Description

Add sequence_pool op

### CODE REVIEW

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc [hunk 1]
```diff
@@ -2773,12 +2773,18 @@ bool RreluOpInferSymbolicShape(pir::Operation *op,
...
+bool SequencePoolOpInferSymbolicShape(
+    pir::Operation *op, pir::InferSymbolicShapeContext *infer_context) {
+  const symbol::ShapeOrDataDimExprs &x_shape_or_data =
+      infer_context->GetShapeOrDataForValue(op->operand_source(0));
+  const std::vector<symbol::DimExpr> &x_shape = x_shape_or_data.shape();
+
+  infer_context->SetShapeOrDataForValue(
+      op->result(0),
+      symbol::ShapeOrDataDimExprs{symbol::TensorShapeOrDataDimExprs(x_shape)});
+
```
Comment (gongshaotian) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc:2787]: 少了个输出
if(pooltype == max){
   ...
}
else{
   infer_context->SetSymbolForValueByStaticShape(op->result(1));
}

**Comment by crazyxiaoxi** [comment]
LGTM

**Comment by crazyxiaoxi** [comment]
> LGTM

补充一下单测状态

**Review by gongshaotian — APPROVED** [review]
LGTM

**Review by Hongqing-work — APPROVED** [review]

**Pull request merged**